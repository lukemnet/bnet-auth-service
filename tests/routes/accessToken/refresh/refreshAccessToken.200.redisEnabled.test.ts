const fastify = require('fastify');
const fp = require('fastify-plugin');
const server = require('../../../../src/index');

describe('/accessToken/refresh (Redis enabled)', () => {
  jest.mock('ioredis');

  const config = {
    app: {
      nodeEnv: 'test',
      port: '8123',
    },
    bnet: {
      region: 'us',
      apiKey: 'key',
      apiSecret: 'secret',
    },
    redis: {
      enable: true,
      connectionString: 'redis://127.0.0.1:6379',
      db: '0',
      replyCachePeriod: 100,
      cacheSegment: 'bas',
    }
  }

  const fastifyServer = fastify();
  fastifyServer.register(server, config);

  beforeEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
  });

  afterAll(async () => {
    await fastifyServer.close();
  });
 
  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/refresh', });
    expect(res.statusCode).toBe(200);
  });


  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/refresh', });
    expect(JSON.parse(res.payload)).toEqual({ status:200, message: "Access token refreshed successfully"});
  });
});