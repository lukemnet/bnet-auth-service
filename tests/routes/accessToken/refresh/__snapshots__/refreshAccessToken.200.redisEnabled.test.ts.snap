// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/accessToken/refresh (Redis enabled) returns correct response 1`] = `"{\\"status\\":200,\\"message\\":\\"Access token refreshed successfully\\"}"`;
